packages/dds/tree/src/test/simple-tree/api/snapshotCompatibilityChecker.spec.ts

```diff
@@ -99,4 +102,64 @@ describe("snapshotCompatibilityChecker", () => {
 		// this means these two versions of the application cannot collaborate on content using these schema.
 		assert.equal(forwardsCompatibilityStatus.canView, false);
 	});
+
+	it("checkCompatibility: allowUnknownOptionalFields", () => {
+		const factory = new SchemaFactoryBeta("test");
+
+		// Point2D is constructed with allowUnknownOptionalFields, so it can read Point3D trees
+		// even though it does not know about the optional field `z`.
+		class Point2D extends factory.object(
+			"Point",
+			{
+				x: factory.number,
+				y: factory.number,
+			},
+			{ allowUnknownOptionalFields: true },
+		) {}
+		class Point3D extends factory.object("Point", {
+			x: factory.number,
+			y: factory.number,
+			z: factory.optional(factory.number),
+		}) {}
+
+		const oldViewSchema = new TreeViewConfiguration({ schema: Point2D });
+		const currentViewSchema = new TreeViewConfiguration({ schema: Point3D });
+
+		// Check to see if a document created with the current view schema can be opened with the historical view schema
+		const backwardsCompatibilityStatus = checkCompatibility(oldViewSchema, currentViewSchema);
+
+		// The current view schema has a superset of the fields on the old view schema, so the schema must be upgraded to add the new
+		// optional field `z`.
+		assert.equal(backwardsCompatibilityStatus.canView, false);
+		assert.equal(backwardsCompatibilityStatus.canUpgrade, true);
+
+		// Test what the old version of the application would do with a tree using the new schema:
+		const forwardsCompatibilityStatus = checkCompatibility(currentViewSchema, oldViewSchema);
+
+		// Content created with the current schema can be viewed by the old schema due to allowUnknownOptionalFields
+		assert.equal(forwardsCompatibilityStatus.canView, true);
+	});
+
+	it("checkCompatibility: staged schema", () => {
+		const factory = new SchemaFactoryBeta("test");
+		const oldSchema = factory.optional(
+			factory.types([numberSchema, factory.staged(stringSchema)]),
+		);
+		const currentSchema = factory.optional(factory.types([stringSchema, numberSchema]));
+
+		const oldViewSchema = new TreeViewConfiguration({ schema: oldSchema });
+		const currentViewSchema = new TreeViewConfiguration({ schema: currentSchema });
+
+		// Check to see if a document created with the current view schema can be opened with the historical view schema
+		const forwardsCompatibilityStatus = checkCompatibility(currentViewSchema, oldViewSchema);
```

Comment:
It does not really matter, but this test and the one above are basically doing the exact same thing for two different kinds of schema changes, so it would be nice if they checked stuff in the same order. Currently one does forwards then backwards compat, and the other test does the backwards compat then forwards. Making them follow the same structure would improve clarity